diff --git a/node_modules/react-native-vision-camera-face-detector/android/src/main/java/com/margelo/nitro/camera/facedetector/HybridImageFaceDetector.kt b/node_modules/react-native-vision-camera-face-detector/android/src/main/java/com/margelo/nitro/camera/facedetector/HybridImageFaceDetector.kt
index 5d116a9..bc96faa 100644
--- a/node_modules/react-native-vision-camera-face-detector/android/src/main/java/com/margelo/nitro/camera/facedetector/HybridImageFaceDetector.kt
+++ b/node_modules/react-native-vision-camera-face-detector/android/src/main/java/com/margelo/nitro/camera/facedetector/HybridImageFaceDetector.kt
@@ -22,15 +22,12 @@ class HybridImageFaceDetector(
   )
 
   private fun resolveInputImage(
-    input: Any?
+    input: InputImage
   ): String {
     return when (input) {
-      is String -> input
-      is Map<*, *> -> {
-        val uri = input["uri"] as? String
-        uri ?: throw IllegalArgumentException("Invalid image object: missing 'uri'")
-      }
-      else -> throw IllegalArgumentException("Invalid image type. Expected string or { uri }")
+      is InputImage.First -> input.value
+      is InputImage.Third -> input.value.uri
+      is InputImage.Second -> throw IllegalArgumentException("Numeric asset references are not supported for static image detection")
     }
   }
 
diff --git a/node_modules/react-native-vision-camera-face-detector/ios/HybridImageFaceDetector.swift b/node_modules/react-native-vision-camera-face-detector/ios/HybridImageFaceDetector.swift
index 608050c..8127bf6 100644
--- a/node_modules/react-native-vision-camera-face-detector/ios/HybridImageFaceDetector.swift
+++ b/node_modules/react-native-vision-camera-face-detector/ios/HybridImageFaceDetector.swift
@@ -52,25 +52,23 @@ class HybridImageFaceDetector: HybridImageFaceDetectorSpec {
   }
 
   private func resolveInputImage(
-    _ input: Any
+    _ input: InputImage
   ) throws -> String {
-    if let uri = input as? String {
-      return uri
-    }
-
-    if let object = input as? [String: Any],
-      let uri = object["uri"] as? String {
+    switch input {
+    case .first(let uri):
       return uri
+    case .second(_):
+      throw NSError(
+        domain: "HybridImageFaceDetector",
+        code: -1,
+        userInfo: [
+          NSLocalizedDescriptionKey:
+          "Numeric asset references are not supported for static image detection"
+        ]
+      )
+    case .third(let imageUri):
+      return imageUri.uri
     }
-
-    throw NSError(
-      domain: "HybridImageFaceDetector",
-      code: -1,
-      userInfo: [
-        NSLocalizedDescriptionKey:
-        "Invalid image type. Expected string or { uri }"
-      ]
-    )
   }
 
   private func createInputImage(
